specs/Freeze.md

```diff
@@ -0,0 +1,155 @@
+```cpp
+bool ViewComponent::HandleWindowMessage(
+    HWND hWnd, UINT message, WPARAM wParam, LPARAM lParam, LRESULT* result)
+{
+    if (message == WM_SYSCOMMAND)
+    {
+        if (wParam == SC_MINIMIZE)
+        {
+            // Hide the webview when the app window is minimized, and freeze it.
+            m_controller->put_IsVisible(FALSE);
+            Freeze();
+        }
+        else if (wParam == SC_RESTORE)
+        {
+            // When the app window is restored, show the webview
+            // (unless the user has toggle visibility off).
+            if (m_isVisible)
+            {
+                Unfreeze();
+                m_controller->put_IsVisible(TRUE);
+            }
+        }
+    }
+}
+
+void ViewComponent::Freeze()
+{
+    HRESULT hr = webView->TryFreeze(
+        Callback<ICoreWebView2StagingTryFreezeCompletedHandler>(
+            [](HRESULT errorCode, BOOL isSuccessful) -> HRESULT {
+                std::wstringstream formattedMessage;
+                formattedMessage << "TryFreeze result (0x" << std::hex << errorCode
+                                 << ") " << (isSuccessful ? "succeeded" : "failed");
+                MessageBox(nullptr, formattedMessage.str().c_str(), nullptr, MB_OK);
+                return S_OK;
+            })
+            .Get());
+    if (FAILED(hr))
+        ShowFailure(hr, L"Call to TryFreeze failed");
+}
+
+void ViewComponent::Unfreeze()
+{
+    wil::com_ptr<ICoreWebView2Staging2> webView;
+    webView = m_webView.query<ICoreWebView2Staging2>();
+    webView->Unfreeze();
+}
+```
+
+# Remarks
+The CoreWebView2Controller must be invisible when the API is called. Otherwise, the
+API fails with `HRESULT_FROM_WIN32(ERROR_INVALID_STATE)`.   
+Freezing is similar to putting a tab to sleep in the Edge browser. Freezing pauses
+WebView script timers and animations, minimizes CPU usage for the associated
+browser renderer process and allows the operating system to reuse the memory that was
+used by the renderer process for other processes.   
+Note that the freeze is best effort and considered completed successfully once the request
+is sent to browser renderer process. If there is a running script, the script will continue
+to run and the renderer process will be frozen after that script is done.
+See [Sleeping Tabs FAQ](https://techcommunity.microsoft.com/t5/articles/sleeping-tabs-faq/m-p/1705434)
+for conditions that might prevent WebView from being frozen. In those situations,
+The TryFreeze operation will fail and the completed handler will be invoked with isSuccessful as false.   
+The WebView will be automatically unfrozen when it becomes visible. Therefore, the app normally doesn't have to call Unfreeze.
+The app can call `Unfreeze` and then `TryFreeze` periodically for an invisibile WebView so that the invisible WebView can sync up with
```

Comment:
```suggestion
The app can call `Unfreeze` and then `TryFreeze` periodically for an invisible WebView so that the invisible WebView can sync up with
```